docs/changelog.rst

```diff
@@ -1,6 +1,61 @@
 Changelog
 =========
 
+Version 0.9 (April 21st, 2023)
+------------------------------
+
+New Features/API
```

Comment:
I'd like to see this reorganized a bit to make it much clearer how much work has been done since the last release and of what kind. Concrete proposal:

- Add a very short paragraph at the top that says roughly "this release adds a huge number of major features and the next major release will be 1.0". 
- Start with a section on major new features and add a one sentence explanation of each (and maybe a link to docs if any). Have one item that mentions the documentation improvements as a whole and refers to the full list further down.
- Then have a section on all the new APIs added.
- Then list the debug mode improvements (to remind people debug mode is awesome).
- Only then list the Python version support changes.